### DISCUSSION TITLE
Adding a non deep learning end to end model training?

### DISCUSSION THREAD

**Discussion by vkullu** [NVIDIA/NVFlare#122] [Q&A] [answered]
Hello,
I am enjoying using your framework. Other frameworks I have come across have been mostly deep learning specific so this is a welcome change!

What would it take to setup a sample example of federated learning for a non deep learning model (like a logistic regression model or better, the recently popular XGBoost or Light GBM models)?

Thank you very much,
Vinayak.

**Comment by yanchengnv** [comment, marked as answer]
Thanks for this good suggestion! 

There are quite a few examples in the "examples" folder already, and some of them have nothing to do with DL (e.g. hello-numpy).

As you might already have noticed, NVFLARE is totally agnostic to what the client is doing (doesn't even have to be machine learning). All it requires is that executors are defined for tasks received from the server, and these executors must return some result (called Shareable).

Tasks are assigned by Controllers running on the Server. We defined a few commonly used controllers (i.e. ScatterAndGather for Fed Average). You can create your own using the Controller API.

As long as the Controllers on the Server and Executors on the Clients are in agreement (tasks issued by the Controllers are executable by executors), you are okay.

That said, we do want to gradually develop more examples for typical use cases (e.g.XGBoost). At the same time, we'd like to encourage users to contribute examples to NVFLARE.

**Reply by vkullu** [reply]
Hi @yanchengnv 
Thank you for the detailed answer. I am having issues adapting your examples to a Keras Multiclass classifier. What is the best way to ask pointed questions? Do you have a Slack/Discord channel ? I need another pair of eyes to run through my relatively straightforward classification example. I can PR it if it seems fit for your examples.
Best,
Vinayak.

**Reply by yanchengnv** [reply]
Thanks Vinayak. The easiest way is PR. Please go ahead to submit your example code to the "examples" folder, and we'll get Keras experts to review and even collaborate together.

NVFLARE is open source and we encourage your participation in creating examples.

Thanks,
Yan

**Comment by YuanTingHsieh** [comment]
@vkullu 
Thanks for the suggestions.

we added more examples other than deep learning now: 

https://github.com/NVIDIA/NVFlare/tree/dev/examples/advanced

k-means, SVM, random forest, ...etc.

Also XGBoost is added as well.

Feel free to try them out!
